src/Native/Unix/System.Security.Cryptography.Native/pal_ssl.c

```diff
@@ -319,6 +324,12 @@ CryptoNative_SslCtxSetCertVerifyCallback(SSL_CTX* ctx, SslCtxSetCertVerifyCallba
     SSL_CTX_set_cert_verify_callback(ctx, callback, arg);
 }
 
+void ResetProtocolRestrictions(SSL_CTX* ctx)
+{
+    SSL_CTX_set_min_proto_version(ctx, 0);
```

Comment:
I didn't get any local failures without adding this to shim (I did for any other API) - presumably it's not needed?

Reply:
CI didn't forgive me though :smile:

Reply:
If you ran `src/Native/build-native.sh` that'll build non-portable, so the shim definition isn't required.  You can also build it portable (`src/Native/build-native.sh -portable`).   CI builds it both ways (we publish portable, RHEL's packages (and other future distro-provided packages) use non-portable).

Reply:
These functions are new in OpenSSL 1.1.0, so they'll need to be FALLBACK_FUNCTION defined.  You can put the body of ResetProtocolRestrictions in `#if NEED_OPENSSL_1_0` to avoid needing to define non-portable equivalence.

FYI: Building on Ubuntu 16.04 will give you OpenSSL 1.0.x headers, 18.04 is 1.1.0 headers, and 19.04 is 1.1.1 headers.

Reply:
I've ended up expanding those macros since they use stuff which we already have available